plugins/paligo.go

```diff
@@ -0,0 +1,335 @@
+		p.Channels.Errors <- fmt.Errorf("error while getting document '%s': %w", item.Name, err)
+		return
+	}
+
+	p.Items <- Item{
+		Content: document.Content,
+		Source:  item.Name,
+		ID:      fmt.Sprint(item.ID),
+	}
+}
+
+/**
+ * Paligo API
+ */
+
+// https://paligo.net/docs/apidocs/en/index-en.html#UUID-a5b548af-9a37-d305-f5a8-11142d86fe20
+const (
+	PALIGO_RATE_LIMIT_CHECK_INTERVAL = 5 * time.Second
+	PALIGO_DOCUMENT_SHOW_LIMIT       = 50
+	PALIGO_FOLDER_SHOW_LIMIT         = 50
+)
+
+func rateLimitPerSecond(rateLimit int) rate.Limit {
+	return rate.Every(time.Minute / time.Duration(rateLimit))
+}
+
+type PaligoItem struct {
+	ID   int    `json:"id"`
+	Name string `json:"name"`
+	UUID string `json:"uuid"`
+	Type string `json:"type"`
+}
+
+type Folder struct {
+	PaligoItem
+	Children []PaligoItem `json:"children"`
+}
+
+type EmptyFolder struct {
+	PaligoItem
+	Children string `json:"children"`
+}
+
+type Component struct {
+	PaligoItem
+	Subtype          string        `json:"subtype"`
+	Creator          int           `json:"creator"`
+	Owner            int           `json:"owner"`
+	Author           int           `json:"author"`
+	CreatedAt        int           `json:"created_at"`
+	ModifiedAt       int           `json:"modified_at"`
+	Checkout         bool          `json:"checkout"`
+	CheckoutUser     string        `json:"checkout_user"`
+	ParentResource   int           `json:"parent_resource"`
+	Taxonomies       []interface{} `json:"taxonomies"`
+	ReleaseStatus    string        `json:"release_status"`
+	Content          string        `json:"content"`
+	Languages        []string      `json:"languages"`
+	External         []interface{} `json:"external"`
+	CustomAttributes []interface{} `json:"custom_attributes"`
+}
+
+type ListFoldersResponse struct {
+	Page       int           `json:"page"`
+	NextPage   string        `json:"next_page"`
+	TotalPages int           `json:"total_pages"`
+	Folders    []EmptyFolder `json:"folders"`
+}
+
+type Document struct {
+	PaligoItem
+	Content   string   `json:"content"`
+	Languages []string `json:"languages"`
+}
+
+type PaligoClient struct {
+	Instance string
+	Username string
+	Token    string
+
+	foldersLimiter   *rate.Limiter
+	documentsLimiter *rate.Limiter
+}
+
+func reserveRateLimit(response *http.Response, lim *rate.Limiter, err error) error {
+	if response.StatusCode != 429 {
+		return err
+	}
+
+	rateLimit := response.Header.Get("Retry-After")
+	if rateLimit == "" {
+		return fmt.Errorf("Retry-After header not found")
+	}
+	seconds, err := strconv.Atoi(rateLimit)
+	if err != nil {
+		return fmt.Errorf("error parsing Retry-After header: %w", err)
+	}
+	log.Warn().Msgf("Rate limit exceeded, need to wait for %d seconds", seconds)
+	lim.SetBurst(1)
+	// We are not waiting for the exact time, because sometimes we get 429 even for fair use,
+	// and sometimes it is released earlier than the specified time
+	time.Sleep(PALIGO_RATE_LIMIT_CHECK_INTERVAL)
```

Comment:
```suggestion
	time.Sleep(seconds + PALIGO_RATE_LIMIT_CHECK_INTERVAL)
```

Reply:
The obvious way is to wait **only** `seconds`.

But the rate limit mechanism in Paligo is not reliable, and after a lot of requests, it may send you a rate limit error with `35 seconds` to wait, but it will actually not reproduce in the next request, after `5 seconds`.

So if you don't want to support this edge case, I think the `time.Sleep` should be only with `seconds`.